### ISSUE TITLE
Long lists not lazily loaded

### ISSUE THREAD

**Issue by meruiden** [lucaszischka/BottomSheet#121]
**Describe the bug**
When using LazyVStack or ForEach, all content gets loaded immediately instead of when they become visible. I can get it to work with a List, but since that doesn't work with ```enableAppleScrollBehavior```, the content scrolls before the list is expanded, completely breaking the scroll effect where the bottom sheet first expands before the content scrolls, so the user would have to use the little anchor to manually expand the sheet.

**Minimal reproduce-able code**
create a bottom sheet with 3 states and ```enableAppleScrollBehavior``` enabled.

contentView:

```swift
LazyVStack(alignment: .leading) {
    ForEach(items, id: \.id) { item in
        Text(item.name)
            .onAppear {
                print("Appeared: \(item.id")
            }
    }
}
```

**Expected behavior**
The content in LazyVStack to call ```onAppear``` only when it becomes visible

**Target version**
- Environment: iOS iPhone 14 pro max simulator
- Version: iOS 16.1 with project targeting iOS 14.0

**Comment by said13** [comment]
Good point. I'm facing the same issue. Did anyone find out how to let it work as expected?

**Comment by alexchernokun** [comment]
@lucaszischka Hello, any updates on this issue? Thanks!

**Comment by lucaszischka** [comment]
`enableAppleScrollBehavior` was quite difficult to implement and buggy from the beginning.
I have rewritten it a couple of times and I haven't found a stable working solution. I might give it a last try sometime, but help is really needed!
